sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5979,9 +6116,49 @@ pi_result piEventRetain(pi_event Event) {
   return PI_SUCCESS;
 }
 
-pi_result piEventRelease(pi_event Event) {
+pi_result piEventReleaseExternal(pi_event Event) {
```

Comment:
What's the difference between `piEventReleaseExternal` and `piEventReleaseInternal`? Just to make sure people pick the right one in the future, could you please make a note somewhere of when to use which?

Reply:
Agree that this is confusing. I inlined piEventReleaseExternal to piEventRelease since it has to be called only from there. Added notes to event release functions: piEventRelease must not be called inside the plugin, piEventReleaseInternal has to be used to release event inside the plugin because piEventRelease  keeps track of external reference count for purpose of this optimization. (similar to piQueueRelease and piQueueReleaseInternal)